todo_app/app.py

```diff
@@ -8,4 +8,77 @@
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    if "todoes" in session:
+        if "removed" in session:
+            return render_template("index.html",todos=session["todoes"],added=session["added"],complete=session["complete"],removed=session["removed"])
+        else:
+            return render_template("index.html",todos=session["todoes"],added=session["added"])
+    else:
+        return render_template("index.html")
+
+@app.route('/add', methods=["POST"])
+def add():
+    if "complete" not in session:
+        session["complete"]=[]
+        complete=[]
+    if "newTask" in request.form:
+        if "todoes" in session:
+            todoes = session["todoes"]
+            todoes.append(request.form["newTask"])
+            session["added"] = [request.form["newTask"]]
+            session["todoes"] = todoes
+        else:
+            session["todoes"] = [request.form["newTask"]]
+            session["added"] = [request.form["newTask"]]
+    if "todoes" not in session:
+        return render_template("index.html")
+    elif "removed" not in session:
+        return render_template("index.html",todos=session["todoes"],added=session["added"],complete=session["complete"])
+    else:
+        return render_template("index.html",todos=session["todoes"],added=session["added"],complete=session["complete"],removed=session["removed"])
```

Comment:
Rather than worrying about what you're displaying from scratch again, we can take advantage of the Flask `redirect` function to let the index handle this case, which has a couple of advantages:
* We only need the logic for the index page in one place, and we can update index if we need to make changes without having to look at what to change elsewhere
* We make the users final request a simple GET request to the index function, which means that for example refreshing the page after adding a card won't try to add a new one (whereas currently it does)
```suggestion
    return redirect('/')
```